File: modules/navigation-trigger/android/src/main/java/expo/modules/navigationtrigger/NavigationTriggerModule.kt
package expo.modules.navigationtrigger

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

class NavigationTriggerModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("NavigationTrigger")

    View(NavigationTriggerView::class) {
      Prop("href") { view: NavigationTriggerView, href: String? ->
        view.setHref(href)
      }
      Prop("type") { view: NavigationTriggerView, type: String? ->
        view.setNavType(type)
      }
      Events("onPress")
    }
  }
}

File: modules/navigation-trigger/ios/NavigationTriggerModule.swift
import ExpoModulesCore

public class NavigationTriggerModule: Module {
  public func definition() -> ModuleDefinition {
    Name("NavigationTrigger")

    View(NavigationTriggerView.self) {
      Prop("href") { (view: NavigationTriggerView, href: String?) in
        view.href = href ?? ""
      }
      Prop("type") { (view: NavigationTriggerView, type: String?) in
        view.navType = type ?? "push"
      }
      Events("onPress")
    }
  }
}
